src/test/java/uk/gov/gchq/palisade/policy/WrappedRuleTest.java

```diff
@@ -45,47 +47,47 @@ public void shouldNotConstruct0Arguments() {
             Assert.fail("exception expected");
         } catch (IllegalArgumentException e) {
             // Then
-            assertEquals("Only one constructor parameter can be non-null", e.getMessage());
+            assertEquals("The IllegalArgumentException should be thrown and show a non null error", "Only one constructor parameter can be non-null", e.getMessage());
         }
     }
 
     @Test
     public void shouldNotConstruct2Arguments() {
         try {
             //when
-            new WrappedRule<>(new TestRule(), o -> o.toString(), null);
+            new WrappedRule<>(new TestRule(), o -> o, null);
             Assert.fail("exception expected");
         } catch (IllegalArgumentException e) {
             //then
-            assertEquals("Only one constructor parameter can be non-null", e.getMessage());
+            assertEquals("The IllegalArgumentException should be thrown and show a non null error", "Only one constructor parameter can be non-null", e.getMessage());
         }
         try {
             //when
             new WrappedRule<>(new TestRule(), null, o -> true);
             Assert.fail("exception expected");
         } catch (IllegalArgumentException e) {
             //then
-            assertEquals("Only one constructor parameter can be non-null", e.getMessage());
+            assertEquals("The IllegalArgumentException should be thrown and show a non null error", "Only one constructor parameter can be non-null", e.getMessage());
         }
         try {
             //when
-            new WrappedRule<>(null, o -> o.toString(), o -> true);
+            new WrappedRule<>(null, (UnaryOperator<Object>) Object::toString, o -> true);
             Assert.fail("exception expected");
         } catch (IllegalArgumentException e) {
             //then
-            assertEquals("Only one constructor parameter can be non-null", e.getMessage());
+            assertEquals("The IllegalArgumentException should be thrown and show a non null error", "Only one constructor parameter can be non-null", e.getMessage());
         }
     }
 
     @Test
     public void shouldNotConstruct3Arguments() {
         try {
             //when
-            new WrappedRule<>(new TestRule(), o -> o.toString(), o -> true);
+            new WrappedRule<>(new TestRule(), o -> o, o -> true);
             Assert.fail("exception expected");
         } catch (IllegalArgumentException e) {
             //then
-            assertEquals("Only one constructor parameter can be non-null", e.getMessage());
+            assertEquals("The IllegalArgumentException should be thrown and show a non null error", "Only one constructor parameter can be non-null", e.getMessage());
```

Comment:
Similar, probably wouldn't inspect the exception message

Reply:
use the @test(expected = xxxException.class) annotation for the same reason as above